Block Navigation: Use quick inserter

noisysocks opened the pull request:
Updates the Block Navigation component to use the quick inserter added in https://github.com/WordPress/gutenberg/pull/22789.

This removes the superfluous category labels and search field when adding a block using Block Navigation in the Navigation screen. It also creates consistency between the two ways to insert blocks on this screen.

Block Navigation in the regular block editor is unaffected by these changes as it does not let you add blocks.

**Before**

<img width="403" alt="Screen Shot 2020-07-07 at 17 38 50" src="https://user-images.githubusercontent.com/612155/86738842-c14ed900-c078-11ea-9e36-47885aacbf1f.png">

**After**

<img width="408" alt="Screen Shot 2020-07-07 at 17 38 30" src="https://user-images.githubusercontent.com/612155/86738850-c3189c80-c078-11ea-88dc-f3dca395368b.png">

noisysocks committed:
Block Navigation: Use quick inserter

shaunandrews commented:
There's some spacing issues around the contents of the inserter. I'd expect it to look more like this:

<img width="353" alt="image" src="https://user-images.githubusercontent.com/191598/86952784-7fc63880-c121-11ea-9b9e-0099e6bd188c.png">

Which makes me wonder if the Quick Library should adapt to its contents; Does it need to have that extra horizontal space? Maybe it could collapse as needed:

<img width="252" alt="image" src="https://user-images.githubusercontent.com/191598/86952862-9b314380-c121-11ea-86f0-f651990eed4f.png">

Similar if there are 4 options, perhaps it uses a 2x2 grid:

<img width="259" alt="image" src="https://user-images.githubusercontent.com/191598/86952900-ab492300-c121-11ea-8dd7-b22004653bb3.png">

noisysocks commented:
@shaunandrews: Thanks for the feedback! Just confirming: This is the same quick inserter that's used in the post editor and in FSE, so if I make those changes then they will apply everywhere. Is that right?

shaunandrews commented:
Correct. I’d expect this behavior to work everywhere. I’d love to know what @Pablo and/or @mtias think.

noisysocks committed:
Quick Inserter: Adjust padding

Remove unnecessary padding when Quick Inserter has no search field or
'Browse all' button.

noisysocks commented:
I removed the extra padding from the Quick Inserter.

<img width="373" alt="Screen Shot 2020-07-14 at 12 59 37" src="https://user-images.githubusercontent.com/612155/87378855-2a32d580-c5d2-11ea-9595-8abf8d505fd3.png">

Note that I made it so that the padding remains when there is a search field or a _Browse all_ button. This looks right to me, but let me know if you disagree and I can remove the padding here too.

<img width="371" alt="Screen Shot 2020-07-14 at 12 58 32" src="https://user-images.githubusercontent.com/612155/87378931-5d756480-c5d2-11ea-9f3e-bdb5e606e0f0.png">

<img width="372" alt="Screen Shot 2020-07-14 at 12 59 19" src="https://user-images.githubusercontent.com/612155/87378934-5ea69180-c5d2-11ea-9d5b-968b31622672.png">

I held off on making the Quick Inserter appear less wide when there are fewer items as it's a slightly more complex change and probably isn't necessary as we'll soon have more blocks which can be added to Navigation.

tellthemachines commented on packages/block-editor/src/components/inserter/style.scss:
```diff
@@ -274,14 +274,14 @@ $block-inserter-tabs-height: 44px;
-		padding: 0;
-		float: left;
-	}
+.block-editor-inserter__quick-inserter:not(.has-search):not(.has-expand) .block-editor-inserter__panel-content {
```
For future-proofing - say at some point we add another class in addition to search and expand, for another variation, and then we're stuck with overriding this super-specific rule - could we not add the wider padding to `.has-search` and `.has-expand` instead, and make this one the default?

noisysocks replied:
Good suggestion! Fixed in dc0a9fdfc1d9e7abf7836adf9cfdb8593e98eb21.

tellthemachines reviewed (commented):
Works well, left a comment below. Also, thanks for fixing that typo 😄

adamziel commented:
Once I select a block, the inserter stays open. I would expect it to go away.

adamziel commented:
> I removed the extra padding from the Quick Inserter.
> 
> <img alt="Screen Shot 2020-07-14 at 12 59 37" width="373" src="https://user-images.githubusercontent.com/612155/87378855-2a32d580-c5d2-11ea-9595-8abf8d505fd3.png">

Not a blocker, but 33% of that inserter is just a blank space. Could it only take as much space as the first two buttons like @shaunandrews suggested?

noisysocks committed:
Quick Inserter: Reduce selector specificity

tellthemachines commented:
> Not a blocker, but 33% of that inserter is just a blank space. Could it only be as wide as the first two buttons like @shaunandrews suggested?

So I actually had a play with this yesterday, and I don't think it's possible unless the container knows about the number of elements inside, which would allow us to attach a class to hook custom styles onto. Otherwise, removing the fixed container width and adjusting the flex layout (or even with a responsive-column grid layout) it'll just default to a single column or, at best, a two column layout, which is undesirable when there are more blocks. And it's only a temporary issue as we're going to have more blocks in there pretty soon 😄

noisysocks committed:
Quick Inserter: Close dropdown when a block is selected

noisysocks commented:
> Once I select a block, the inserter stays open. I would expect it to go away.

Fixed in 2f5a6598ac3712caf2608253a2ddc7da8a3acf52.

> So I actually had a play with this yesterday, and I don't think it's possible unless the container knows about the number of elements inside, which would allow us to attach a class to hook custom styles onto.

I was able to get it nearly working by setting the width of `block-editor-block-types-list__list-item` to `( $block-inserter-width - $padding ) / 3` and then using `display: inline-block` instead of flex. But setting `$padding` correctly is a mess because it's different in the quick inserter versus the regular inserter. 

Like you say, I don't think it's worth it given this is a temporary problem 🙂

tellthemachines reviewed (approved):
Looks good!
